Sources/Rendering/WebGPU/Types/index.js

```diff
@@ -0,0 +1,74 @@
+// ----------------------------------------------------------------------------
+// vtkWebGPUDevice static functions
+//
+// WebGPU uses types in a many places and calls, and often those types
+// need to be associated with byte sizes, alignments, native arrays etc.
+// The folowing methods are designed to help vtk.js introspect those types.
+// WebGPU currently tends to use multiple type formats:
+//  - buffer types such as float32x4
+//  - shader types suchs as vec4<f32>
+//  - texture types such as rgba32float
+// ----------------------------------------------------------------------------
+
+function getByteStrideFromBufferFormat(format) {
+  if (!format) return 0;
+  let numComp = 1;
+  if (format.substring(format.length - 2) === 'x4') numComp = 4;
+  if (format.substring(format.length - 2) === 'x3') numComp = 3;
+  if (format.substring(format.length - 2) === 'x2') numComp = 2;
+
+  let typeSize = 4;
+  if (numComp > 1) {
+    if (format.substring(format.length - 3, format.length - 1) === '8x') {
+      typeSize = 1;
+    }
+    if (format.substring(format.length - 4, format.length - 1) === '16x') {
+      typeSize = 2;
+    }
+  } else {
+    if (format.substring(format.length - 1) === '8') typeSize = 1;
+    if (format.substring(format.length - 2) === '16') typeSize = 2;
+  }
+  return numComp * typeSize;
```

Comment:
Hum it event looks like it could be written like 

```
return format.split('x').map(Number).reduce((a,b) => a*b, 1/8);
```

Reply:
That doesn't work because an input like float32x4 splits to float32, 4 and float32 fails Number. 

I'll rework it to be a bit shorter and do less heap allocs.

Reply:
I see, but if not called often, the overhead is not the end of the world. ;-)

You can use some pre-compiled regexp for the extraction of those 2 numbers ;-)